include/clink/feature/one_hot_encoder.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLINK_FEATURE_ONE_HOT_ENCODER_H_
+#define CLINK_FEATURE_ONE_HOT_ENCODER_H_
+
+#include "clink/feature/proto/one_hot_encoder.pb.h"
+#include "clink/linalg/sparse_vector.h"
+
+namespace clink {
+
+// Params of OneHotEncoderModel.
+typedef struct OneHotEncoderParams {
+  // Whether to drop the last category.
+  bool isDropLast;
+} OneHotEncoderParams;
+
+// A Model which encodes data into one-hot format.
+class OneHotEncoderModel {
+public:
+  // Converts the provided data to a SparseVector.
+  SparseVector transform(int x, int columnIndex);
+
+  // Loads a OneHotEncoderModel from given path.
+  static OneHotEncoderModel *load(std::string path);
+
+private:
```

Comment:
Can we follow Google's C++ guide for `Copyable and Movable Types` specified in [1]? It says "A class's public API must make clear whether the class is copyable, move-only, or neither copyable nor movable".

For best performance, in general we prefer move-only constructor unless there is explicit use-case for copy-constructor.

The following code snippet would ensure only the move constructor is supported for this class.

```
// Move operations are supported.
OneHotEncoderModel(OneHotEncoderModel&& other);

OneHotEncoderModel& operator=(OneHotEncoderModel&& other);

// This class is not copyable or assignable.
OneHotEncoderModel(const OneHotEncoderModel& other) = delete;
OneHotEncoderModel& operator=(const OneHotEncoderModel&) = delete;
```

[1] https://google.github.io/styleguide/cppguide.html#Copyable_Movable_Types

Reply:
I'll make the change. What's more, in TFRT classes also follow this convention to define whether they are movable or copyable, but few of them provide comments like `// Move operations are supported.`. I think we should also make it a must to add such comments.